Automattic/woocommerce-subscriptions-core
# Refactor get_recurring_cart_key() so it's available for grouping products into subscriptions

## Description

This PR refactors the `get_recurring_cart_key()` function so the functionality which groups subscription products/cart items into subscriptions with the same billing terms can be utilised in grouping order items into subscriptions. This change is necessary for a new REST API endpoint that will create subscriptions from parent order line items. See https://github.com/woocommerce/woocommerce-subscriptions/issues/4562

## How to test this PR

This PR should result in no direct change/impact. 

1. Add various subscription products with different billing schedules to your cart. eg
   - Period - weekly, monthly, daily, yearly
   - Intervals - every 2nd x, every 4th y
   - Free trials
   - Lengths/end dates
   - Synced
2. Confirm that subscription products are grouped correctly on the cart and checkout pages into recurring carts.

## Product impact
